src/App.tsx

```diff
@@ -1,61 +1,127 @@
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './Types/Todo';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { getUserById } from './Helpers/getUser';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './Types/User';
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+  const [title, setTitle] = useState<string>('');
+  const [userId, setUserId] = useState<number | string>('');
+  const [isTitleValid, setIsTitleValid] = useState<boolean>(true);
+  const [isUserSelected, setIsUserSelected] = useState<boolean>(true);
+
+  const todosWithUsers: Todo[] = todos.map(todo => ({
+    ...todo,
+    user: getUserById(todo.userId, usersFromServer),
+  }));
+
+  const addTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title || !userId) {
+      if (!title) {
+        setIsTitleValid(false);
+      }
+
+      if (!userId) {
+        setIsUserSelected(false);
+      }
+
+      return;
+    }
+
+    const maxId = Math.max(...todos.map(todo => todo.id));
+
+    const user = getUserById(+userId, usersFromServer);
+
+    if (user) {
+      setTodos((current => [
+        ...current,
+        {
+          id: maxId + 1,
+          userId: user.id,
+          title,
+          completed: false,
+          user,
+        },
+      ]));
+    }
```

Comment:
You render users in select from `usersFromServer`. 
Is it possible situation when you don't have a user ?

```suggestion
      const newTodo = {
          id: getNewTodoId(todos),
          userId: userId,
          title,
          completed: false,
          user: getUserById(+userId, usersFromServer),
      };
      setTodos((current => [ ...current, newTodo ]));
```

Reply:
fixed 👌
well, I don't why, but my previous attempt to do it without conditional checking was failed by typescript on Find method possibly returning undefined...
can't understand why it is working now